test/unit/update_and_kill_test.lua

```diff
@@ -0,0 +1,97 @@
+local fiber = require("fiber")
+local t = require("luatest")
+
+local helpers = require("test.helper")
+
+local g = t.group('update_and_kill', {
+    {index_type = 'TREE', engine = 'vinyl'},
+    {index_type = 'TREE', engine = 'memtx'},
+    {index_type = 'HASH', engine = 'memtx'},
+})
+
+g.before_each({index_type = 'TREE'}, function(cg)
+    t.skip_if(cg.params.engine == 'vinyl' and not helpers.vinyl_is_supported(),
+        'Blocked by https://github.com/tarantool/tarantool/issues/6448')
+    cg.first_space = helpers.create_space_with_tree_index(cg.params.engine)
+    local second_space = helpers.create_space('second_space', cg.params.engine)
+    second_space:create_index('primary')
+    cg.second_space = second_space
+end)
+
+g.before_each({index_type = 'HASH'}, function(cg)
+    cg.first_space = helpers.create_space_with_hash_index(cg.params.engine)
+    local second_space = helpers.create_space('second_space', cg.params.engine)
+    second_space:create_index('primary', {type = 'HASH'})
+    cg.second_space = second_space
+end)
+
+g.after_each(function(cg)
+    if cg.task ~= nil then
+        cg.task:kill()
+    end
```

Comment:
There is no cg.task in the tests, but it could be a list with all tasks `cg.tasks`.